### PR TITLE
[Shopsys] Add parallel and regenerate to Gitlab pipelines

### PR DESCRIPTION
**Pull request by AmpMVn**
#### Description, the reason for the PR
It was not possible to manually run a cypress test regeneration in Gitlab UI. I added a regeneration pipeline that can be run manually in Gitlab UI plus added the ability to set up parallel runs.

#### Check the appropriate checkboxes below, please

- [ ] [BC breaks](https://docs.shopsys.com/en/latest/contributing/backward-compatibility-promise/) 
- [ ] New feature 
- [x] I have read and signed [License Agreement for contributions](https://www.shopsys.com/license-agreement)

----
:globe_with_meridians: Live Preview:
  - https://mvn-ssp-2844-gitlab-parallel-cypress-tests-generate.odin.shopsys.cloud
  - https://cz.mvn-ssp-2844-gitlab-parallel-cypress-tests-generate.odin.shopsys.cloud

### CODE REVIEW

File: project-base/.gitlab-ci.yml [hunk 1]
```diff
@@ -134,12 +143,65 @@ test:storefront-acceptance:
...
+        - docker compose exec -T php-fpm php phing -D production.confirm.action=y composer-prod db-create frontend-api-generate-new-keys build-demo-dev-quick error-pages-generate
+        - GROUP=$GROUP docker compose run cypress-regenerate-screenshots
+        - git add ./storefront/cypress/snapshots/e2e
+        - git status
+        - |
+            if ! git diff --staged --quiet; then
+                git commit -m "Regenerated Cypress screenshots for ${GROUP} group"
+                git fetch origin "$CI_COMMIT_REF_NAME"
+                git pull --rebase origin "$CI_COMMIT_REF_NAME"
+                git push https://shopsys_bot:$API_TOKEN@gitlab.shopsys.cz/ss6-projects/project-base.git "$CI_COMMIT_REF_NAME" --force -v || (echo "Failed to push changes" && git log -n 5 && exit 1)
```
Comment (vitek-rostislav) [inline project-base/.gitlab-ci.yml]:
```suggestion
                git push https://shopsys_bot:$API_TOKEN@gitlab.shopsys.cz/ss6-projects/$CI_PROJECT_NAME.git "$CI_COMMIT_REF_NAME" --force -v || (echo "Failed to push changes" && git log -n 5 && exit 1)
```

**Review by vitek-rostislav — CHANGES_REQUESTED** [review]
please, add the upgrade notes with link project base diff for this PR and also for https://github.com/shopsys/shopsys/pull/3514

File: project-base/.gitlab-ci.yml [hunk 2]
```diff
@@ -134,12 +143,65 @@ test:storefront-acceptance:
...
+        - git checkout "$CI_COMMIT_REF_NAME"
+        - cp -f ./gitlab/docker-compose-ci.yml ./docker-compose.yml
+        - docker compose up -d
+        - docker compose exec -T php-fpm php phing -D production.confirm.action=y composer-prod db-create frontend-api-generate-new-keys build-demo-dev-quick error-pages-generate
+        - GROUP=$GROUP docker compose run cypress-regenerate-screenshots
+        - git add ./storefront/cypress/snapshots/e2e
+        - git status
+        - |
+            if ! git diff --staged --quiet; then
+                git commit -m "Regenerated Cypress screenshots for ${GROUP} group"
```
Comment (AmpMVn) [inline project-base/.gitlab-ci.yml]:
```suggestion
                if [ -z "$GROUP" ]; then
                    git commit -m "Regenerated Cypress screenshots"
                else
                    git commit -m "Regenerated Cypress screenshots for ${GROUP} group"
                fi
```

File: project-base/.gitlab-ci.yml [hunk 3]
```diff
@@ -134,12 +143,69 @@ test:storefront-acceptance:
...
+        - |
+            if ! git diff --staged --quiet; then
+                if [ -z "$GROUP" ]; then
+                    git commit -m "Regenerated Cypress screenshots"
+                else
+                    git commit -m "Regenerated Cypress screenshots for ${GROUP} group"
+                fi
+                git fetch origin "$CI_COMMIT_REF_NAME"
+                git pull --rebase origin "$CI_COMMIT_REF_NAME"
+                git push https://shopsys_bot:$API_TOKEN@gitlab.shopsys.cz/ss6-projects/project-base.git "$CI_COMMIT_REF_NAME" --force -v || (echo "Failed to push changes" && git log -n 5 && exit 1)
```
Comment (AmpMVn) [inline project-base/.gitlab-ci.yml]:
```suggestion
                git push https://shopsys_bot:$API_TOKEN@gitlab.shopsys.cz/ss6-projects/${CI_PROJECT_NAME}.git "$CI_COMMIT_REF_NAME" --force -v || (echo "Failed to push changes" && git log -n 5 && exit 1)
```

**Review by vitek-rostislav — APPROVED** [review]

**Review by pk16011990 — APPROVED** [review]

**Pull request merged**